### PR TITLE
DRIVERS-2435 add option to enable `featureFlagFLE2ProtocolVersion2` 

### PR DESCRIPTION
**Pull request by kevinAlbs**
# Summary

- Enable `featureFlagFLE2ProtocolVersion2` if environment variable `ENABLE_featureFlagFLE2ProtocolVersion2=ON` is set in `run-orchestration.sh`

Updated scripts were tested in Go driver tests: https://github.com/mongodb/mongo-go-driver/pull/1213

# Background & Motivation

Enabling `featureFlagFLE2ProtocolVersion2` is required to enable the Queryable Encryption v2 protocol. [SERVER-69563](https://jira.mongodb.org/browse/SERVER-69563) plans to enable the feature flag by default. Once the feature flag is enabled by default, the `setfle2parameter.py` script and check for `ENABLE_featureFlagFLE2ProtocolVersion2` can be removed.

### CODE REVIEW

**Comment by kevinAlbs** [comment]
@katcharov I am unable to request you as a reviewer. Can you review? This is a dependency for https://github.com/mongodb/specifications/pull/1391

I requested membership to the mongodb-labs organization for your GitHub user.

**Review by eramongodb — APPROVED** [review]
Minor suggestions; otherwise, LGTM.

File: .evergreen/run-orchestration.sh [hunk 1]
```diff
@@ -83,6 +83,16 @@ fi
 
 perl -p -i -e "s|ABSOLUTE_PATH_REPLACEMENT_TOKEN|${DRIVERS_TOOLS}|g" $ORCHESTRATION_FILE
 
+if [ "$ENABLE_featureFlagFLE2ProtocolVersion2" = "ON" -a "$MONGODB_VERSION" = "latest" ]; then
+  # This is a temporary workaround until featureFlagFLE2ProtocolVersion2 is enabled by default in SERVER-69563. Once latest server builds have SERVER-69563, this if block may be removed.
```
[the remark this answers is not in the document]
Comment (kevinAlbs) [reply .evergreen/run-orchestration.sh:87]: Created and noted [DRIVERS-2590](https://jira.mongodb.org/browse/DRIVERS-2590)

File: .evergreen/run-orchestration.sh [hunk 2]
```diff
@@ -83,6 +83,16 @@ fi
...
+  echo "rewrite orchestration config to add setParameter featureFlagFLE2ProtocolVersion2=1 ... begin"
+  . "$DIR/find-python3.sh"
+  PYTHON="$(find_python3)"
```
[the remark this answers is not in the document]
Comment (kevinAlbs) [reply .evergreen/run-orchestration.sh]: Good point. The  the setfle2parameter.py script only needed minor tweaks to work with Python 2.7. Use of `find_python3` has been removed.

**Comment by katcharov** [comment]
@kevinAlbs LGTM!

**Pull request merged**